### DISCUSSION TITLE
"Internal Server Error"  responsed by nominatim python frontend serve

### DISCUSSION THREAD

**Discussion by ghost** [osm-search/Nominatim#3482] [General]
I installed the 4.4.0 in my local follow the Guide. The map data of China have been import into postgresql.  
I test the Nominatim serve with python falcon by "nominatim serve" and all the query are success.  
Then I deploy  with the python frontend by setting up Nominatim as a systemd job and configuring nginx.  The status and reverse query by the deployed service are success, but the search request reponsed {"title": "500 Internal Server Error"}.  
I checked the deployed request log by python falcon, and found that there are only the reverse request and the search requests are not exist.

Example query:  `curl "http://localhost:8088/search.php?q=%E5%8C%97%E4%BA%AC&polygon_geojson=0"`
8088 is used by Test serve   and 8070 is used by Deploy serve
The response of Test serve:
`[{"place_id":2593890,"licence":"Data © OpenStreetMap contributors, ODbL 1.0. http://osm.org/copyright","osm_type":"node","osm_id":25248662,"lat":"39.9057136","lon":"116.3912972","category":"place","type":"city","place_rank":15,"importance":0.7344014383682882,"addresstype":"city","name":"北京市","display_name":"北京市, 东城区, 北京市, 100010, 中国","boundingbox":["39.7457136","40.0657136","116.2312972","116.5512972"]},{"place_id":626570,"licence":"Data © OpenStreetMap contributors, ODbL 1.0. http://osm.org/copyright","osm_type":"relation","osm_id":912940,"lat":"40.190632","lon":"116.412144","category":"boundary","type":"administrative","place_rank":8,"importance":0.7344014383682882,"addresstype":"state","name":"北京市","display_name":"北京市, 中国","boundingbox":["39.1707096","41.0595584","115.4172086","117.7371243"]}]`

The response of Deployed serve:
`{"title": "500 Internal Server Error"}`

Anyone has the same problem? How can I fix it?

**Comment by mtmail** [comment]
By "Deploy serve" you mean the gunicorn + nginx setup described in https://nominatim.org/release-docs/latest/admin/Deployment-Python/ ?

Most likely it's a missing installed dependency or a dependency cannot be found in the default Python library search path. The gunicorn log file or `StandardOutput=append:/var/log/gunicorn-nominatim.log` will contain the full error message.
